src/cert_store.rs

```diff
@@ -63,11 +70,95 @@ impl CertStore {
+    pub fn system(which: &str) -> io::Result<CertStore> {
+        unsafe {
+            let data = OsStr::new(which)
+                             .encode_wide()
+                             .chain(Some(0))
+                             .collect::<Vec<_>>();
+            let store = crypt32::CertOpenStore(winapi::CERT_STORE_PROV_SYSTEM_W as winapi::LPCSTR,
+                                               0,
+                                               0,
+                                               winapi::CERT_SYSTEM_STORE_CURRENT_USER,
+                                               data.as_ptr() as *mut _);
+            if store.is_null() {
+                Err(io::Error::last_os_error())
+            } else {
+                Ok(CertStore(store))
+            }
+        }
+    }
+
+    /// Imports a PKCS#12-encoded key/certificate pair, returned as a
+    /// `CertStore` instance.
+    ///
+    /// The password must also be provided to decrypt the encoded data.
+    pub fn import_pkcs12(data: &[u8], password: &str) -> io::Result<CertStore> {
+        unsafe {
+            let mut blob = winapi::CRYPT_INTEGER_BLOB {
+                cbData: data.len() as winapi::DWORD,
+                pbData: data.as_ptr() as *mut u8,
+            };
+            let password = OsStr::new(password)
+                                 .encode_wide()
+                                 .chain(Some(0))
+                                 .collect::<Vec<_>>();
+            let res = crypt32::PFXImportCertStore(&mut blob,
+                                                  password.as_ptr(),
+                                                  0);
+            if res.is_null() {
+                Err(io::Error::last_os_error())
+            } else {
+                Ok(CertStore(res))
+            }
+        }
+    }
+
+    /// Returns an iterator over the certificates in this certificate store.
+    pub fn iter(&mut self) -> CertStoreIter {
+        CertStoreIter { store: self, cur: None }
+    }
 }
 
+/// An iterator over the certificates contained in a `CertStore`, returned by
+/// `CertStore::iter`
+pub struct CertStoreIter<'a> {
+    store: &'a mut CertStore,
+    cur: Option<CertContext>,
+}
+
+impl<'a> Iterator for CertStoreIter<'a> {
+    type Item = CertContext;
+
+    fn next(&mut self) -> Option<CertContext> {
+        unsafe {
+            let cur = self.cur.as_ref().map(|p| p.as_inner());
+            let cur = cur.unwrap_or(ptr::null_mut());
+            let next = crypt32::CertEnumCertificatesInStore(self.store.0, cur);
+
+            if next.is_null() {
+                self.cur = None;
+                None
+            } else {
+                let next = CertContext::from_inner(next);
+                self.cur = Some(next.clone());
+                Some(next)
+            }
+        }
+    }
+}
+
+/// An in-memory store of certificates and CTLs, created by `CertStore::memory`
+/// and can be converted into a `CertStore`.
+#[derive(Clone)]
```

Comment:
Maybe it could just be called `CertStore` and dropped entirely? I'm not sure if there are restrictions on calling the two methods here on arbitrary `CertStore` types, but the windows apis at least allow it!